sympy/matrices/utilities.py

```diff
@@ -20,6 +20,25 @@
 # everywhere and setting to `True` will turn it on everywhere in matrices where
 # it can be applied.
 
+# There are a few other places in the matrix code where dotprodsimp can probably
+# help, these are places where a call is made to:
+#
+# dps = _get_intermediate_simp()
+#
+# To determine whether dotprodsimp helps in these places testing needs to be
+# done, to turn dotprodsimp on in these places by default replace this call with:
+#
+# from sympy.simplify.simplify import dotprodsimp as _dotprodsimp
+# dps = _get_intermediate_simp(_dotprodsimp)
+#
+# Or possibly:
+#
+# from sympy import expand_mul
+# dps = _get_intermediate_simp(expand_mul, expand_mul)
+#
+# This second form uses lighter simplification by default but may still do
+# better than nothing.
+
 # True, False or None
 _DOTPRODSIMP_MODE = False if os.environ.get('SYMPY_DOTPRODSIMP', '').lower() in \
         ('false', 'off', '0') else None
```

Comment:
Can this be written using an ordinary `if/else`?

I still don't see the need for this environment variable. At some point we will just want to get rid of it.

Reply:
Yes it can, and yes you can get rid of this variable if you really want. I'm just afraid of getting the blame for some user's important code breaking without recourse. Maybe give it one live cycle and then kill it?

Reply:
Given that multiply does not use dotprodsimp by default, what scenario would lead to breakage (I'm guessing you still had to fix some things)?

Reply:
Yes some tests did change, mostly due to functions dependent on `rref`. But those are under our control, we don't know what user code out there depends on. I would rather that some important app can get back up and running immediately just by setting an environment variable while they adapt it for the new structure of results. Besides, you only need to mention this if someone complains, and in any case you remove it for the version after, its just a safety.

Reply:
For this to be useful it needs to be mentioned in the release notes I guess. We should probably edit them generally after all these changes. You can see them here:
https://github.com/sympy/sympy/wiki/Release-Notes-for-1.6

At the moment there are a bunch of notes that are not really relevant to end users like:
```
Pulled all MatrixEigen function implementations out of matrices/matrices.py into a separate
matrices/eigen.py source file for easier maintenance and future matrix class overhaul. (#18425
by @Pristine-Cat)
```
It would be better to remove those and have a note explaining the relevant points like what behaviour has changed, how you can use the environment variable, what new functions and function parameters there are etc.

Reply:
So basically go through all those PR's, remove irrelevant release notes and add somewhere a single or a few notes explaining all matrix changes?

Reply:
It's not necessary to go through the PRs. Once a PR is merged the release note is transferred to the wiki. Subsequent changes can be made directly on the wiki (care is needed to preserve formatting so that the sympy-bot can continue to transfer release notes over). The wiki remains the primary source of release notes after release.

I'm not sure who exactly has permission to edit the wiki...